File: sdk/android/silentpass/src/main/java/com/silentpass/sdk/SilentPassConfig.kt
package com.silentpass.sdk

/**
 * Configuration for the SilentPass SDK.
 *
 * @param baseURL API base URL
 * @param apiKey API key for authentication
 * @param appID Application identifier
 * @param timeoutMs Request timeout in milliseconds
 * @param sandbox Enable sandbox mode for development
 */
data class SilentPassConfig(
    val baseURL: String = "https://api.silentpass.io",
    val apiKey: String,
    val appID: String,
    val timeoutMs: Long = 15_000,
    val sandbox: Boolean = false,
)

File: sdk/ios/Sources/SilentPass/SilentPassConfig.swift
import Foundation

/// Configuration for the SilentPass SDK.
public struct SilentPassConfig {
    /// API base URL.
    public let baseURL: String
    /// API key for authentication.
    public let apiKey: String
    /// Application identifier.
    public let appID: String
    /// Request timeout in seconds.
    public let timeout: TimeInterval
    /// Enable sandbox mode (logs debug info).
    public let sandbox: Bool

    public init(
        baseURL: String = "https://api.silentpass.io",
        apiKey: String,
        appID: String,
        timeout: TimeInterval = 15,
        sandbox: Bool = false
    ) {
        self.baseURL = baseURL
        self.apiKey = apiKey
        self.appID = appID
        self.timeout = timeout
        self.sandbox = sandbox
    }
}
